### ISSUE TITLE
list_fields() always returns NULL

### ISSUE THREAD

**Issue by regisoc** [AlexIoannides/elasticsearchr#53]
Hi,

I am trying to use the functions you listed in [CRAN](https://cran.r-project.org/web/packages/elasticsearchr/vignettes/quick_start.html), everything seems to work pretty fine except the `list_fields()` using `%info%` infix.

I tried with the tidyverse `mpg` dataset to play small.

![image](https://user-images.githubusercontent.com/25327259/66520092-0d63f180-ead8-11e9-98c4-d6c79e502b20.png)

I use the version 7.4.0 of Elasticsearch (the latest). 

Do I miss something?

**Comment by regisoc** [comment]
Found it. 

In the file `R/api.R`, `list_fields()` function, line 189:

```r
fields <- names(index_mapping[[1]]$mappings$data$properties)
```

must be changed by:

```r
fields <- names(index_mapping[[1]]$mappings$properties)
```

When performed:
![image](https://user-images.githubusercontent.com/25327259/66655819-4f9e4780-ec2c-11e9-87d9-87b4108e71ce.png)

**Comment by regisoc** [comment]
pull request sent, however I forgot to mention: **it only works with ES 7**.

Taking the same dataset as example (`mpg` from `tidyverse`), all the minor versions of ES have differences : 

With **ES 7.4.0**: executing `curl http://localhost:9200/mpg/_mapping?pretty`

```json
{
  "mpg" : {
    "mappings" : {
      "properties" : {
        "class" : {
          "type" : "text",
          "fields" : {
            "keyword" : {
              "type" : "keyword",
              "ignore_above" : 256
            }
          }
        },
        "cty" : {
          "type" : "long"
        },
   ...
}
```
The hierarchy to reach fields name is: `mpg > mappings > properties`

With **ES 6.8.3**: executing `curl http://localhost:9200/mpg/_mapping?pretty`

```json
{
  "mpg" : {
    "mappings" : {
      "mpg" : {
        "properties" : {
          "class" : {
            "type" : "text",
            "fields" : {
              "keyword" : {
                "type" : "keyword",
                "ignore_above" : 256
              }
            }
          },
  ....
}
```
The hierarchy to reach fields name is: `mpg > mappings > mpg > properties`

**Comment by AlexIoannides** [comment]
Yes, some serious development (together with a cogent plan), is required to make `elasticsearchr` compatible with ES 7.x.
